x-pack/plugins/ml/public/application/services/elastic_models_service.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type ModelDefinitionResponse } from '@kbn/ml-trained-models-utils';
+import { type TrainedModelsApiService } from './ml_api_service/trained_models';
+
+export type ElserVersion = 1 | 2;
+
+export class ElasticModels {
+  constructor(private readonly trainedModels: TrainedModelsApiService) {}
+
+  /**
+   * Provides an ELSER config
```

Comment:
If this is the function we're expecting other teams to call then it would be nice to have a precise statement of what it does.

For example, if somebody asks for version 2 and they're on Cloud do they get the Linux x86_64 optimized version or the portable version? (They _should_ get the Linux x86_64 optimized version in this situation.)

Reply:
The comment has been updated [here](https://github.com/elastic/kibana/pull/167000/commits/298e20139cb82e1e50845846266d8e7e195bb680).
 Also pushed the [fix](https://github.com/elastic/kibana/pull/167000/commits/bd117fb48c1c4febbd01173ab3989c6e56380366) for returning a recommended model based on the requested version.